pgscatalog_utils/download/download_scorefile.py

```diff
@@ -1,185 +1,111 @@
 
 def download_scorefile() -> None:
     args = _parse_args()
-    set_logging_level(args.verbose)
+    config.set_logging_level(args.verbose)
     _check_args(args)
-    _mkdir(args.outdir)
-
-    if args.build is None:
-        logger.warning(f'Downloading scoring file(s) in the author-reported genome build')
-    elif args.build in ['GRCh37', 'GRCh38']:
-        logger.warning(f'Downloading harmonized scoring file(s) in build: {args.build}.')
-    else:
-        logger.critical(f'Invalid genome build specified: {args.build}. Only -b GRCh37 and -b GRCh38 are supported')
-        raise Exception
 
-    overwrite_existing_file = args.overwrite_existing_file
-    if overwrite_existing_file:
+    build: typing.Union[None, GenomeBuild]
+    match args.build:
+        case None:
+            logger.info("Downloading scoring file(s) in the author-reported genome build")
+            build = None
+        case "GRCh37":
+            build = GenomeBuild.GRCh37
+        case "GRCh38":
+            build = GenomeBuild.GRCh38
+        case _:
+            logger.critical(f"Invalid genome build specified: {args.build}")
+            logger.critical("Only -b GRCh37 and -b GRCh38 are supported")
+            raise Exception
+
+    if build is not None:
+        logger.info(f"Downloading harmonised scoring file(s) in build: {build}")
+
+    if args.overwrite_existing_file:
         logger.debug("--overwrite, overwriting new version of the Scoring file, if available")
-        logger.warning(f'Existing Scoring files will be overwritten if new versions of the Scoring files are available for download.')
+        logger.warning(
+            "Existing Scoring files will be overwritten if new versions of the Scoring files are available for download.")
 
-    pgs_lst: list[list[str]] = []
-
-    pgsc_calc_info = None
     if args.pgsc_calc:
-        pgsc_calc_info = args.pgsc_calc
+        config.PGSC_CALC_VERSION = args.pgsc_calc_info
+
+    config.OUTDIR = pathlib.Path(args.outdir).resolve()
+    logger.info(f"Download directory: {config.OUTDIR}")
+    config.OUTDIR.mkdir(exist_ok=True)
+    config.OVERWRITE = args.overwrite_existing_file
 
+    results: list[list[CatalogResult]] = []
     if args.efo:
         if args.efo_include_children:
             logger.debug("--trait set, querying traits (including PGS for child terms)")
+            for term in args.efo:
+                results.append(CatalogQuery(CatalogCategory.TRAIT, term, include_children=True,
+                                            pgsc_calc_version=config.PGSC_CALC_VERSION).get())
         else:
             logger.debug("--trait set, querying traits")
-        pgs_lst = pgs_lst + [query_trait(x, pgsc_calc_info, args.efo_include_children) for x in args.efo]
-
+            for term in args.efo:
+                results.append(CatalogQuery(CatalogCategory.TRAIT, term, include_children=False,
+                                            pgsc_calc_version=config.PGSC_CALC_VERSION).get())
```

Comment:
Seems a bit redundant: could we set a temp variable for the `include_children` boolean, e.g.:
```
        inc_child = False
        if args.efo_include_children:
            logger.debug("--trait set, querying traits (including PGS for child terms)")
            inc_child = True
        else:
            logger.debug("--trait set, querying traits")
        results.append(CatalogQuery(CatalogCategory.TRAIT, term, include_children=inc_child,
                                  pgsc_calc_version=config.PGSC_CALC_VERSION).get())

```